sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonStreamingSourceRunner.scala

```diff
@@ -164,7 +175,20 @@ class PythonStreamingSourceRunner(
       val pickledPartition: Array[Byte] = PythonWorkerUtils.readBytes(dataIn)
       pickledPartitions.append(pickledPartition)
     }
-    pickledPartitions.toArray
+    val prefetchedRecordsStatus = dataIn.readInt()
+    val iter: Option[Iterator[InternalRow]] =
+      if (prefetchedRecordsStatus == NON_EMPTY_PYARROW_RECORD_BATCHES) {
```

Comment:
[nit] a case match might be easier to read.

Reply:
Nice catch! Changed to pattern match.